table/block_based/block_based_table_builder.cc

```diff
@@ -1313,6 +1318,14 @@ void BlockBasedTableBuilder::BGWorkWriteRawBlock() {
     }
 
     r->pc_rep->file_size_estimator.SetCurrBlockRawSize(block_rep->data->size());
+
+    if (r->table_options.prepopulate_block_cache) {
```

Comment:
It might be cleaner to just pass the uncompressed contents also to ```WriteRawBlock``` and let it handle the insertion into both the caches.